src/olympia/addons/tasks.py

```diff
@@ -536,28 +536,13 @@ def flag_high_hotness_according_to_review_tier():
         # Need a hotness threshold to be set for the tier.
         growth_threshold_before_flagging__isnull=False,
     )
-    base_qs = Addon.unfiltered.exclude(status=amo.STATUS_DISABLED).filter(
-        type=amo.ADDON_EXTENSION
-    )
+    # Build a single queryset with all add-ons we want to flag for each tier.
+    base_qs = UsageTier.get_base_addons()
     hotness_per_tier_filters = Q()
     for usage_tier in usage_tiers:
-        tier_filters = {
-            'average_daily_users__gte': usage_tier.lower_adu_threshold,
-            'average_daily_users__lt': usage_tier.upper_adu_threshold,
-        }
-        mean_hotness_in_tier = (
-            base_qs.filter(**tier_filters)
-            .aggregate(Avg('hotness', default=0))
-            .get('hotness__avg')
-        )
-        if not mean_hotness_in_tier:
-            continue
-        hotness_ceiling_before_flagging = mean_hotness_in_tier * (
-            1 + usage_tier.growth_threshold_before_flagging / 100
-        )
```

Comment:
Old algorithm was defined here

Reply:
I guess I'm wondering about the algorithm change? If the issue was about penalizing addons in tiers with negative average growth, could we invert the behaviour depending on if the average growth is above or below 0?


After talking it through with GPT something like this?

```python
def calculate_hotness(mean_hotness_in_tier, growth_threshold):
    if mean_hotness_in_tier > 0:
        return mean_hotness_in_tier * (1 + growth_threshold)
    elif mean_hotness_in_tier < 0:
        return mean_hotness_in_tier * (1 / (1 + growth_threshold))
    else:
        return 0
```

This would mean as the negative growth approaches zero we reduce the threshold to zero and as positive growth increases we increase it to infinity, I think.. 🙃 🤓

Reply:
I originally had a variant of this using `abs()` but Operations / @abhn explicitly requested to use percentage points instead (https://mozilla-hub.atlassian.net/browse/ADDONSOPS-381).

Reply:
Not trying to block, but does the function I shared above not handle zero correctly? Maybe you could describe what was incorrect about zero growth in a tier in the PR context?

Reply:
Your suggestion would have `0` as the threshold if the average hotness in the tier was `0`, which is undesirable. As I said, this was discussed with Operations and they decided, after having looked at the data, to go for a percentage point increase (an addition) instead of a percentage increase (a multiplication).